dev/tools/controllerbuilder/pkg/codegen/typegenerator.go

```diff
@@ -116,7 +118,7 @@ func (g *TypeGenerator) writeVisitedMessages() {
 
 		k := generatedFileKey{
 			GoPackage: g.goPackage,
-			FileName:  "types.generated.go",
+			FileName:  strings.ToLower(g.goKind) +".types.generated.go",
```

Comment:
I was under the impression we'd have a single 'types.generated.go' shared across multiple resources. Did I get that wrong?

Reply:
types.generated.go will get clobbered by the most recent type IIUC. The intent here is to keep these separate and the cmd user can select how they deal with this file

Reply:
> types.generated.go will get clobbered by the most recent type IIUC

To fix this issue, can we modify the flags to pass in a list of protos? That way, we can generate all the types for a set of protos in the same API group, instead of just one at a time.

Reply:
> That way, we can generate all the types for a set of protos in the same API group

Or maybe we can pass in _just_ the API group. The issue w a list of protos is what happens, say, 6 months after we added resources A and B. Someone wants to add resource C, they then have to recall to add both resources that came before to their generation cmd which is a bit of a paper cut imo.

Reply:
I thought that was the purpose of https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/dev/tools/controllerbuilder/generate.sh ? To record what had been run before. And, eventually, we could run it regularly.